charms/tensorboard-controller/tests/integration/test_charm.py

```diff
@@ -0,0 +1,200 @@
+    """Deploy Istio Ingress Gateway and Istio Pilot."""
+    await ops_test.model.deploy(
+        entity_url="istio-gateway",
+        application_name=istio_gateway,
+        channel="latest/edge",
+        config={"kind": "ingress"},
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        istio_pilot,
+        channel="latest/edge",
+        config={"default-gateway": "test-gateway"},
+        trust=True,
+    )
+    await ops_test.model.add_relation(istio_pilot, istio_gateway)
+
+    await ops_test.model.wait_for_idle(
+        apps=[istio_pilot, istio_gateway],
+        status="active",
+        raise_on_blocked=False,
+        timeout=60 * 20,
+    )
+
+
+async def test_istio_gateway_info_relation(ops_test: OpsTest):
+    """Setup Istio and relate it to the Tensorboard Controller."""
+    # setup Istio
+    istio_gateway = "istio-ingressgateway"
+    istio_pilot = "istio-pilot"
+    await _setup_istio(ops_test, istio_gateway, istio_pilot)
+
+    # add Tensorboard-Controller/Istio relation
+    await ops_test.model.add_relation(f"{istio_pilot}:gateway-info", f"{APP_NAME}:gateway-info")
+
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
+    )
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=2, min=1, max=10),
+    stop=tenacity.stop_after_attempt(30),
+    reraise=True,
+)
+def assert_replicas(client, resource_class, resource_name, namespace):
+    """Test for replicas.
+
+    Retries multiple times to allow for tensorboard to be created (i.e. readyReplicas == 1).
+    """
+    tensorboard = client.get(resource_class, resource_name, namespace=namespace)
+    replicas = tensorboard.get("status", {}).get("readyReplicas")
+
+    resource_class_kind = resource_class.__name__
+    if replicas == 1:
+        logger.info(f"{resource_class_kind}/{resource_name} readyReplicas == {replicas}")
+    else:
+        logger.info(
+            f"{resource_class_kind}/{resource_name} readyReplicas == {replicas} (waiting for '1')"
+        )
+
+    assert replicas == 1, f"Waited too long for {resource_class_kind}/{resource_name}!"
+
+
+async def test_create_tensorboard(ops_test: OpsTest):
+    """Test Tensorboard creation."""
+    lightkube_client = Client()
+    load_in_cluster_generic_resources(lightkube_client)
+
+    # Create PVC for Tensorboard logs and Tensorboard
+    resources = codecs.load_all_yaml(
+        "\n---\n".join([PVC_TEMPLATE_FILE.read_text(), TENSORBOARD_TEMPLATE_FILE.read_text()]),
+        context={"pvc_name": PVC_NAME, "tensorboard_name": TENSORBOARD_NAME},
+    )
+    for rsc in resources:
+        lightkube_client.create(rsc, namespace=ops_test.model_name)
+
+    try:
+        tensorboard_ready = lightkube_client.get(
+            TENSORBOARD_RESOURCE,
+            name=TENSORBOARD_NAME,
+            namespace=ops_test.model_name,
+        )
+    except ApiError:
+        assert False
```

Comment:
Does this try/except add clarity to the test suite?  If you catch the error and then `assert False`, I think you lose all the details from the error's traceback.   If the test should immediately fail, why not just let the error be uncaught?

Reply:
Serious case of copy pasting there. I didn't have the chance to play around with the lightkube client TBH, so I don't really know what error I'm expecting there. Quite frankly, I feel that either this or the next assert is redundant.